ports/zephyr/zephyr_storage.c

```diff
@@ -139,6 +139,22 @@ MP_DEFINE_CONST_OBJ_TYPE(
 #endif // CONFIG_DISK_ACCESS
 
 #ifdef CONFIG_FLASH_MAP
+
+#define DT_DRV_COMPAT fixed_partitions
+
+#define FLASH_AREA_DEFINE_LABEL(part) CONCAT(MP_QSTR_, DT_STRING_TOKEN(part, label))
+#define FLASH_AREA_DEFINE_NB(part) CONCAT(MP_QSTR_, DT_FIXED_PARTITION_ID(part))
+
+#define FLASH_AREA_DEFINE_GETNAME(part) COND_CODE_1(DT_NODE_HAS_PROP(part, label), \
+    (FLASH_AREA_DEFINE_LABEL(part)), (FLASH_AREA_DEFINE_NB(part)))
+
+#define FLASH_AREA_DEFINE_DEFINE(part) { MP_ROM_QSTR(FLASH_AREA_DEFINE_GETNAME(part)), MP_ROM_INT(DT_FIXED_PARTITION_ID(part)) },
+
+#define FLASH_AREA_DEFINE(part) COND_CODE_1(DT_NODE_HAS_STATUS_OKAY(DT_MTD_FROM_FIXED_PARTITION(part)), \
+    (FLASH_AREA_DEFINE_DEFINE(part)), ())
+
+#define FOREACH_PARTITION(n) DT_FOREACH_CHILD(DT_DRV_INST(n), FLASH_AREA_DEFINE)
```

Comment:
We can do without DT_DRV_COMPAT pretty easily. Just change this to 

```
#define FOREACH_PARTITION(n) DT_FOREACH_CHILD(DT_INST(n, fixed_partitions), FLASH_AREA_DEFINE)
```

and remove DT_DRV_COMPAT. I think this makes things actually easier to understand.

Reply:
This looks better yea, will look into it

Reply:
That does not work, but i figured another slightly different solution